cmd/ctl/audit/audit.go

```diff
@@ -100,6 +101,15 @@ func runAuditCmd(clientGen clientGenFunc) func(*cobra.Command, []string) error {
 			return err
 		}
 
+		r, err := client.ServerStatus(context.Background(), &requestv1.ServerStatusRequest{})
+		if err != nil {
+			return err
+		}
+
+		if r.Version != util.Version {
+			return fmt.Errorf("server version and client version does not match: server: %s, client: %s", r.Version, util.Version)
+		}
```

Comment:
Is this necessarily a problem? The API should remain backward and forward compatible at least between minor and patch versions.

Reply:
Yeah actually we should do a semantic version check here. This was for demonstration purposes. BTW, do we provide the forward compatibility? I mean we should be able to add new api methods w/p bumping the major version right? In that case it should be handy to have a version check. At least we can warn the users.

Reply:
I see, what do you think of having a server status method for the API? Should we keep it anyway or go lean? :) BTW, we can use this utility to raise a warning for some possible inconsistent behaviors.

Reply:
We can keep the `ServerStatus` RPC (but maybe rename to `ServerInfo` or something similar because "status" has a specific meaning in this context). I would remove the explicit version check for now and maybe change the output of the `version` flag to provide both the server and client API versions (if the server is online). Again, `kubectl` is a good example to follow but we probably don't need to output as much information as they do.

```
Client Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.0", GitCommit:"cb303e613a121a29364f75cc67d3d580833a7479", GitTreeState:"clean", BuildDate:"2021-04-08T16:31:21Z", GoVersion:"go1.16.1", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.7", GitCommit:"132a687512d7fb058d0f5890f07d4121b3f0a2e2", GitTreeState:"clean", BuildDate:"2021-05-12T12:32:49Z", GoVersion:"go1.15.12", Compiler:"gc", Platform:"linux/amd64"}
```

Reply:
Makes sense 👍 Going to update accordingly.